airflow/providers/microsoft/azure/hooks/wasb.py

```diff
@@ -140,9 +140,13 @@ def get_conn(self) -> BlobServiceClient:  # pylint: disable=too-many-return-stat
             return BlobServiceClient(account_url=f"https://{conn.login}.blob.core.windows.net/" + sas_token)
 
         # Fall back to old auth (password) or use managed identity if not provided.
+        credential = conn.password
+        if not credential:
+            credential = ManagedIdentityCredential()
+            self.log.info("Using managed identity as credential")
         return BlobServiceClient(
             account_url=f"https://{conn.login}.blob.core.windows.net/",
-            credential=conn.password or ManagedIdentityCredential(),
+            credential=conn.password,
```

Comment:
Well. For me this one looks like it REMOVES the MIC support  from what I seem the previous code worked in the way this PR intended to (if conn.password = None, then managedidentity credential was created and used. This PR makes the credential created but not used. Did I miss something?

Reply:
Should it be

```suggestion
            credential=credential,
```

But then it is just adding an extra log entry in case the MIC is used.

Reply:
Of course – fixed. Nice catch